pkg/common/kubeutils/apitoken_provider.go

```diff
@@ -0,0 +1,35 @@
+package kubeutils
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// APITokenProvider wraps around the kubernetes interface to enhance testability
+type APITokenProvider struct {
+	clientSet kubernetes.Interface
+}
+
+// NewAPITokenProvider creates new APITokenProvider
+func NewAPITokenProvider(useInClusterConfig bool) (*APITokenProvider, error) {
+	clientSet, err := GetClientSet(useInClusterConfig)
+	if err != nil {
+		return nil, fmt.Errorf("Could not create APITokenProvider: %s", err.Error())
+	}
+	return &APITokenProvider{clientSet: clientSet}, nil
+}
+
+// GetKeptnAPITokenFromSecret returns the `keptn-api-token` data secret from Keptn Installation
+func (a *APITokenProvider) GetKeptnAPITokenFromSecret(namespace string, secretName string) (string, error) {
```

Comment:
should we change all the methods that are using client set internally to accept the context as first param?